java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -0,0 +1,250 @@
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JObject extends AbstractJSONSchema2Pojo {
+
+    private static final String JAVA_UTIL_MAP = "java.util.Map";
+    private static final String ADDITIONAL_PROPERTIES = "additionalProperties";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JObject.class);
+    private static final Set<String> IGNORED_FIELDS = new HashSet<>();
+
+    static {
+        IGNORED_FIELDS.add("description");
+        IGNORED_FIELDS.add("schema");
+        IGNORED_FIELDS.add("example");
+        IGNORED_FIELDS.add("examples");
+    }
+
+    private String type = null;
+    private Map<String, AbstractJSONSchema2Pojo> fields = new HashMap<>();
+    private Set<String> required = new HashSet<>();
+    private JObjectOptions options;
+
+    public JObject(
+            String type,
+            Map<String, JSONSchemaProps> fields,
+            List<String> required,
+            JObjectOptions options) {
+        this.options = options;
+
+        if (required != null) {
+            this.required.addAll(required);
+        }
+
+        String nextPrefix = options.getPrefix();
+        String nextSuffix = options.getSuffix();
+
+        if (type.toLowerCase(Locale.ROOT).equals("spec")) {
+            nextPrefix = "";
+            nextSuffix = "Spec";
+        }
+
+        this.type =
+                AbstractJSONSchema2Pojo.sanitizeString(
+                        options.getPrefix()
+                                + type.substring(0, 1).toUpperCase()
+                                + type.substring(1)
+                                + options.getSuffix());
+
+        if (fields == null) {
+            // no fields
+        } else {
+            for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
+                if (!IGNORED_FIELDS.contains(field.getKey()))
+                    this.fields.put(
+                            field.getKey(),
+                            AbstractJSONSchema2Pojo.fromJsonSchema(
+                                    field.getKey(), field.getValue(), nextPrefix, nextSuffix));
+            }
+        }
+    }
+
+    @Override
+    public String getType() {
+        return this.type;
+    }
+
+    @Override
+    public GeneratorResult generateJava(CompilationUnit cu) {
+        ClassOrInterfaceDeclaration clz = cu.getClassByName(this.type).orElse(null);
+
+        if (clz != null) {
+            // TODO: investigate a more nested structure for the generated code
+            LOGGER.warn(
+                    "A class named {} have been already processed, if this class have multiple implementations the resulting code might be incorrect",
```

Comment:
s/have/has/

Reply:
done, I have the code ready to remove this though